internal/metrics/metrics.go

```diff
@@ -114,13 +125,14 @@ func Init(ctx context.Context, r Reporter, interval time.Duration) func() {
 		observables...,
 	)
 	if err != nil {
-		log.Fatalf("failed to register callback: %v", err)
+		log.Error(err, "failed to register callback")
+		os.Exit(2)
```

Comment:
Nit: I would stick with a single non-zero exit code unless we have this documented, have constants defined, and have a reason to have multiple.